app/src/main/java/org/mozilla/fenix/settings/quicksettings/ConnectionPanelDialogFragment.kt

```diff
@@ -0,0 +1,165 @@
+                        findViewById<View>(com.google.android.material.R.id.design_bottom_sheet) as FrameLayout
+                    val behavior = BottomSheetBehavior.from(bottomSheet)
+                    behavior.state = BottomSheetBehavior.STATE_EXPANDED
+                }
+            }
+        } else {
+            Dialog(requireContext()).applyCustomizationsForTopDialog(inflateRootView())
+        }
+    }
+
+    private fun Dialog.applyCustomizationsForTopDialog(rootView: View): Dialog {
+        addContentView(
+            rootView,
+            LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.MATCH_PARENT
+            )
+        )
+
+        window?.apply {
+            setGravity(args.gravity)
+            setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+            // This must be called after addContentView, or it won't fully fill to the edge.
+            setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
+        }
+        return this
+    }
+
+    private fun navigateToQuickSettings() {
+        getCurrentTab()?.let { tab ->
+            requireComponents.useCases.trackingProtectionUseCases.containsException(tab.id) { contains ->
+                dismiss()
+                val isTrackingProtectionEnabled = tab.trackingProtection.enabled && !contains
+                val directions =
+                    BrowserFragmentDirections.actionGlobalQuickSettingsSheetDialogFragment(
+                        sessionId = tab.id,
+                        url = tab.content.url,
+                        title = tab.content.title,
+                        isSecured = tab.content.securityInfo.secure,
+                        sitePermissions = args.sitePermissions,
+                        gravity = args.gravity,
+                        certificateName = tab.content.securityInfo.issuer,
+                        permissionHighlights = tab.content.permissionHighlights,
+                        isTrackingProtectionEnabled = isTrackingProtectionEnabled
+                    )
+                findNavController().nav(R.id.quickSettingsSheetDialogFragment, directions)
+            }
+        }
+    }
+
+    internal fun getCurrentTab(): SessionState? {
+        return requireComponents.core.store.state.findTabOrCustomTab(args.sessionId)
+    }
+
+    /**
+     * Construct an initial [WebsiteInfoState] to be rendered by [WebsiteInfoView]
+     * based on the current website's status and connection.
+     *
+     * While being displayed users have no way of modifying it.
+     *
+     * @param websiteUrl [String] the URL of the current web page.
+     * @param websiteTitle [String] the title of the current web page.
+     * @param isSecured [Boolean] whether the connection is secured (TLS) or not.
+     * @param certificateName [String] the certificate name of the current web  page.
+     */
+    internal fun createWebsiteInfoState(
```

Comment:
Maybe we should move this to where `WebsiteInfoState` is defined since this functionality isn't necessarily tied to the fragment. I also see we have `QuickSettingsFragmentStore.createWebsiteInfoState` - ideally we try to avoid this duplication.

Reply:
👍🏽